### PR TITLE
gerardog.gsudo: add portable version to 2.4.2

### PR DESCRIPTION
**Pull request by soredake**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/131849)

### CODE REVIEW

**Comment by soredake** [comment]
Fixed file name.

**Comment by stephengillie** [comment]
Hi @soredake,

Manual Validation ended with: 
> 10:22:38 AM: This command cannot be run due to the error: The specified executable is not a valid application for this OS platform.

Running the EXE opened this dialog:
image

Should the application be interacted with in a different way?

File: manifests/g/gerardog/gsudo/portable/2.4.2/gerardog.gsudo.portable.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: gerardog.gsudo.portable
+PackageVersion: 2.4.2
+InstallerLocale: en-US
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: net46-AnyCpu/gsudo
+  PortableCommandAlias: gsudo
```
Comment (gerardog) [inline manifests/g/gerardog/gsudo/portable/2.4.2/gerardog.gsudo.portable.installer.yaml:11]: Should have been:

- RelativeFilePath: net46-AnyCpu/gsudo.exe
  PortableCommandAlias: gsudo.exe

Comment (soredake) [reply manifests/g/gerardog/gsudo/portable/2.4.2/gerardog.gsudo.portable.installer.yaml]: I initially tried to link shell script for wsl, but that did not go well as winget symlinks everything as .exe.

**Comment by soredake** [comment]
`./` In relative path is needed until https://github.com/microsoft/winget-cli/issues/2884 is fixed.

**Review by Trenly — CHANGES_REQUESTED** [review]
Why isn't this being added into `geardog.gsudo` as a new installer node? Having it as a separate package could cause issues with correlation - see all of the [Area-Matching](https://github.com/microsoft/winget-cli/issues?q=is%3Aopen+is%3Aissue+label%3AArea-Matching) issues over at CLI when packages are too similar

Users will be able to use `--installer-type portable` to specify they want the portable install

**Comment by soredake** [comment]
> Users will be able to use `--installer-type portable` to specify they want the portable install

Oh, i didn't know about this option, will make changes later.

**Comment by soredake** [comment]
@gerardog now per-arch binaries are used.

@Trenly i've added portable versions to main manifest.

**Comment by gerardog** [comment]
How does this portable flag works? Is there any way I can test this PR before merging?

**Comment by Trenly** [comment]
> How does this portable flag works? Is there any way I can test this PR before merging?

You can use `winget install --manifest <path_to_manifest> --installer-type portable` to specify you want the portable installer and not the msi installer

**Comment by stephengillie** [comment]
This PR omits these files that are present in the current manifest:
>  

(Automated response - build 513.)

**Comment by stephengillie** [comment]
Hi @soredake and @gerardog,

Should this be a separate package, with PackageIdentifier `gerardog.gsudo.portable` as suggested by the title? Or are you happy with adding the portable packages to the existing `gerardog.gsudo` package, as this manifest change performs?

**Comment by Trenly** [comment]
> Hi @soredake and @gerardog,
> 
> Should this be a separate package, with PackageIdentifier `gerardog.gsudo.portable` as suggested by the title? Or are you happy with adding the portable packages to the existing `gerardog.gsudo` manifest, as this change performs?

@stephengillie - I suggested previously that it all be one package due to the `--installer-type` switch now being available; It seems the title was just not updated

**Review by stephengillie — APPROVED** [review]

**Pull request merged**